nixos/modules/services/misc/gotenberg.nix

```diff
@@ -0,0 +1,258 @@
+          PDF engines to enable. Each one can be used to perform a specific task.
+          See [the documentation](https://gotenberg.dev/docs/configuration#pdf-engines) for more details.
+          Defaults to all possible PDF engines.
+        '';
+      };
+
+      logLevel = mkOption {
+        type = types.enum [
+          "error"
+          "warn"
+          "info"
+          "debug"
+        ];
+        default = "info";
+        description = "The logging level for Gotenberg.";
+      };
+
+      environmentFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = "Environment file to load extra environment variables from.";
+      };
+
+      extraArgs = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        description = "Any extra command-line flags to pass to the Gotenberg service.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.enableBasicAuth -> cfg.environmentFile != null;
+        message = ''
+          When enabling HTTP Basic Authentication with `services.gotenberg.enableBasicAuth`,
+          you must provide an environment file via `services.gotenberg.environmentFile` with the appropriate environment variables set in it.
+
+          See `services.gotenberg.enableBasicAuth` for the names of those variables.
+        '';
+      }
+    ];
+
+    systemd.services.gotenberg = {
+      description = "Gotenberg API server";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      path = [ cfg.package ];
+      environment = {
+        LIBREOFFICE_BIN_PATH = "${cfg.libreoffice.package}/lib/libreoffice/program/soffice.bin";
+        CHROMIUM_BIN_PATH = lib.getExe cfg.chromium.package;
+        FONTCONFIG_FILE = pkgs.makeFontsConf {
+          fontDirectories = [ pkgs.liberation_ttf_v2 ] ++ cfg.extraFontPackages;
+        };
+      };
+      serviceConfig = {
+        Type = "simple";
+        DynamicUser = true;
+        ExecStart = "${lib.getExe cfg.package} ${lib.escapeShellArgs args}";
+
+        # Hardening options
+        PrivateDevices = true;
+        PrivateIPC = true;
+        PrivateUsers = true;
+
+        ProtectClock = true;
+        ProtectControlGroups = true;
+        ProtectHome = true;
+        ProtectHostname = true;
+        ProtectKernelLogs = true;
+        ProtectKernelModules = true;
+        ProtectKernelTunables = true;
+        ProtectProc = "invisible";
+        ProcSubset = "pid";
+
+        RestrictAddressFamilies = [
+          "AF_UNIX"
+          "AF_INET"
+          "AF_INET6"
+          "AF_NETLINK"
+        ];
+        RestrictNamespaces = true;
+        RestrictRealtime = true;
+
+        LockPersonality = true;
+        MemoryDenyWriteExecute = true;
+
+        SystemCallFilter = [
+          "@system-service"
+          "~@resources"
+          "~@privileged"
+        ];
+        SystemCallArchitectures = "native";
```

Comment:
Note that for me, when using this module imported as `unstable` into NixOS 24.05, `gotenberg` crashes with `SIGSYS`:

```
Started Gotenberg API server.
gotenberg.service: Main process exited, code=dumped, status=31/SYS
```

Of course it might be due to my backport, but not sure what the difference would be.

I see

```
kernel: audit: type=1326 audit(1729094920.959:2): auid=4294967295 uid=61254 gid=61254 ses=4294967295 subj=kernel pid=3469490 comm=".gotenberg-wrap" exe="/nix/store/ny6ni5yyj4fy95hf9vsj6xk7nalgjcv0-gotenberg-8.8.1/bin/.gotenberg-wrapped" sig=31 arch=c000003e syscall=160 compat=0 ip=0x407d2e code=0x80000000
```

Looks like syscall 160 is `setrlimit()`.

That's forbidden by `"~@resources"`, is that intended?

Reply:
This makes it work for me as a workaround, without changing the module:

```nix
systemd.services.gotenberg.serviceConfig.SystemCallFilter = lib.mkAfter ["@resources"];
```

Reply:
this should have been resolved with #344770, though it may have not been backported.

Reply:
Ah got it, my `unstable` pin was slightly too old.

Thanks for already fixing it!